src/hotspot/share/opto/doCall.cpp

```diff
@@ -233,7 +234,8 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
       }
       if (receiver_method == NULL &&
           (have_major_receiver || morphism == 1 ||
-           (morphism == 2 && UseBimorphicInlining))) {
+           (morphism == 2 && UseBimorphicInlining) ||
+           (morphism >= 2 && PolymorphicInlining))) {
         // receiver_method = profile.method();
         // Profiles do not suggest methods now.  Look it up in the major receiver.
         receiver_method = callee->resolve_invoke(jvms->method()->holder(),
```

Comment:
```
next_hit_cg = this->call_generator(next_receiver_method,
                                  vtable_index, !call_does_dispatch, jvms,
                                  allow_inline, prof_factor);
              if (next_hit_cg != NULL && !next_hit_cg->is_inline() &&
                  have_major_receiver && UseOnlyInlinedBimorphic) {
                  // Skip if we can't inline second receiver's method
                  next_hit_cg = NULL;
```
I think we only need try to allow inline when mophisim is exactly 2. 
`allow_inline && mophisim == 2`

Reply:
done